src/devices/device_models/common/add_variable.jl

```diff
@@ -12,8 +12,10 @@ function add_variable(ps_m::CanonicalModel,
    ps_m.variables[var_name] = _container_spec(ps_m.JuMPmodel, (PSY.get_name(d) for d in devices), time_steps)
```

Comment:
@jd-lara could we do this in the below for loop? It seems like we are looping through the devices twice...

Reply:
The `()` is just a generator it actually doesn't allocate any memory. The issue here is that we need the axis to allocate the array before looping.

Reply:
It does eventually have to be evaluated (iterate through the loop), so if there is a way to do it in the following for loop, that would be better.

Reply:
Is this code even performance-sensitive?

Reply:
I can look into it but it is not an obvious solution. The only way is to evaluate it inside of the macro and specify it in a way that the container is created inside of the macro.

Reply:
> Is this code even performance-sensitive?

The allocation of the Array isn't the creation of the @constraint it is. The only way out of this double call is to make it inside on a long macro call. I haven't looked into it since the update to JuMP v0.19.